conf/template/icub-config-build-tree.cmake.in

```diff
@@ -4,21 +4,30 @@
 
 if (NOT ICUB_FOUND)
 
-message(STATUS "Using iCub from build tree")
+  message(STATUS "Using iCub from build tree")
 
-set(ICUB_VERSION @ICUB_VERSION@)
+  set(ICUB_VERSION @ICUB_VERSION@)
+  # If YARP_LIBRARIES has been valorized by a previous call of find_package(YARP)
+  # in user CMakelists.txt save it in temporary variable to restore it later.
+  if(YARP_LIBRARIES)
+    set(_temp_yarp_libs ${YARP_LIBRARIES})
+  endif()
 
-include("@CMAKE_BINARY_DIR@/@EXPORT_INCLUDE_FILE@")
-include("@CMAKE_BINARY_DIR@/@EXPORT_CONFIG_FILE@")
+  find_package(YARP COMPONENTS OS conf sig dev math
+                    OPTIONAL_COMPONENTS gsl)
```

Comment:
This does not convince me, either you require gsl (if it is required by any iCub target) or not. Having this optional could lead to some weird error if YARP was found but YARP_gsl is not available, I think that the error issued by CMake telling you that the component is missing is way more informative than the linking error that will be issued by the compiler...

Reply:
So do you suggest to put 
```cmake
find_dependency(YARP 3.0 REQUIRED COMPONENTS OS conf sig dev math gsl)
```
Even if maybe `YARP_gsl` is not needed?
It is used only by `learningMachine` that it is not compiled if YARP_gsl_FOUND is set to true.